module/button.py

```diff
@@ -1,18 +1,67 @@
 from telegram import Update
 from telegram.ext import CallbackContext
+from module.add_book import add_book
+from module.add_item import add_item
 from module.create_connection import create_connection
-from module.shared import DB_PATH
+from module.shared import DB_PATH, error_message
+from module.add_book import add_book
+from module.add_item import add_item
+from module.manage_requests import delete_request, update_similar_requests
 
 
 def button(update: Update, context: CallbackContext) -> None:
     chat_id = update.effective_chat.id
     query = update.callback_query
-    conn = create_connection(DB_PATH)
-    if conn:
+    
+    if query.data.split(';')[0] == 'delete':
+        conn = create_connection(DB_PATH)
+        if conn:
+            cur = conn.cursor()
+            q = query.data.split(';')[1]
+            cur.execute("DELETE FROM Market WHERE rowid=?", (q,))
+            query.edit_message_text(text=f"Eliminazione del libro selezionato...")
+            conn.commit()
+            context.bot.send_message(chat_id, "Libro eliminato.")
+        else:
+            query.edit_message_text(text=f"Si è verificato un problema nella lettura del database.")
+        conn.close()
+    
+    if query.data.split(';')[0] == 'new_request':
+        vote = query.data.split(';')[1]
+        row_id = query.data.split(';')[2]
```

Comment:
avoid splitting multiple times the same data using the destructuring/unpack feature
```suggestion
        _, vote, row_id = query.data.split(';')[1]
```